packages/rs-dapi-client/src/platform/get_documents.rs

```diff
@@ -0,0 +1,186 @@
+//! `GetDocuments*` requests.
+
+use dapi_grpc::platform::v0::{self as platform_proto, Proof, ResponseMetadata};
+use drive::query::{OrderClause, StartClause, WhereClause};
+
+use crate::{platform::IncompleteMessage, transport::TransportRequest, DapiRequest, Settings};
+
+fn to_proto_where_clause(
+    where_clause: WhereClause,
+) -> platform_proto::get_documents_request::WhereClause {
+    let mut value_cbor_bytes = Vec::new();
+    ciborium::into_writer(&where_clause.value, &mut value_cbor_bytes)
+        .expect("byte slice is a safe writer");
+
+    platform_proto::get_documents_request::WhereClause {
+        field: where_clause.field,
+        operator: where_clause.operator as i32,
+        value: value_cbor_bytes,
+    }
+}
+
+fn to_proto_order_clause(
+    order_clause: OrderClause,
+) -> platform_proto::get_documents_request::OrderClause {
+    platform_proto::get_documents_request::OrderClause {
+        field: order_clause.field,
+        ascending: order_clause.ascending,
+    }
+}
+
+fn to_proto_start(start_clause: StartClause) -> platform_proto::get_documents_request::Start {
+    use platform_proto::get_documents_request::Start as ProtoStart;
+    match start_clause {
+        StartClause::StartAt(x) => ProtoStart::StartAt(x.to_vec()),
+        StartClause::StartAfter(x) => ProtoStart::StartAfter(x.to_vec()),
+    }
+}
+
+/// Request documents bytes.
+#[derive(Debug)]
+pub struct GetDocuments {
+    /// Data contract ID
+    pub data_contract_id: Vec<u8>,
+    /// Document type for the data contract
+    pub document_type: String,
+    /// `where` clauses for the query
+    pub where_clauses: Vec<WhereClause>,
+    /// `order_by` clauses for the query
+    pub order_by_clauses: Vec<OrderClause>,
+    /// queryset limit
+    pub limit: u32,
+    /// TODO start
+    pub start: Option<StartClause>,
+}
+
+impl DapiRequest for GetDocuments {
+    type DapiResponse = GetDocumentsResponse;
+
+    const SETTINGS_OVERRIDES: Settings = Settings::default();
+
+    type Error = IncompleteMessage;
+
+    type TransportRequest = platform_proto::GetDocumentsRequest;
+
+    fn to_transport_request(&self) -> Self::TransportRequest {
+        platform_proto::GetDocumentsRequest {
```

Comment:
As far as I understand what  @QuantumExplorer  said, the user should generate GetDocumentsRequest from DriveQuery, not create it manually.

Reply:
1. Does it mean I should make `limit` an option? And then just send a default to gRPC?

2. Unfortunately, I have no clear requirements on this task; we can add conversions later if needed, but in any case `DriveQuery` is not something DAPI Client shall execute as is